okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -60,6 +62,16 @@
 public class Platform {
   public static final Logger logger = Logger.getLogger(Platform.class.getName());
 
+  /**
+   * List of security providers to use in order of preference.
+   */
+  private static final String[] ANDROID_SECURITY_PROVIDERS = new String[]{
+      // See https://developer.android.com/training/articles/security-gms-provider.html
+      "com.google.android.gms.org.conscrypt.OpenSSLProvider",
+      "com.android.org.conscrypt.OpenSSLProvider",
+      "org.conscrypt.OpenSSLProvider",
```

Comment:
I think we should err on the side of known good behavior over trying to infer what the developer intent was from that API. I think it's too indirect to be reliable. If developers want more direct control over what GRPC does they should set the socket factory on the builder.